examples/faerie.rs

```diff
@@ -0,0 +1,172 @@
+use cranelift::codegen::ir::types;
+use cranelift::prelude::*;
+use cranelift_faerie::{FaerieBackend, FaerieBuilder, FaerieTrapCollection};
+use cranelift_module::{Linkage, Module};
+
+fn main() {
+    // The Faerie backend only supports ISAs with PIC so we add a flag to say it's enabled.
+    let mut flag_builder = settings::builder();
+    flag_builder.enable("is_pic").unwrap();
+
+    // We use the native platform this program is running on.
+    let isa_builder = cranelift_native::builder().unwrap();
+    // We create the ISA we will target.
+    let isa = isa_builder.finish(settings::Flags::new(flag_builder));
+
+    // Set up a builder using our target ISA.
+    let builder = FaerieBuilder::new(
+        isa,
+        String::from("sum"),
+        FaerieTrapCollection::Disabled,
+        cranelift_module::default_libcall_names(),
+    )
+    .unwrap();
+
+    // Set up a module using the FaerieBackend which outputs object files for the selected ISA.
+    let mut module: Module<FaerieBackend> = Module::new(builder);
+
+    // The context holds the state of the current code generation.
+    let mut context = module.make_context();
+
+    // Creating a function signature of `fn(i32, i32) -> i32`.
+    context
+        .func
+        .signature
+        .params
+        .push(AbiParam::new(types::I32));
+    context
+        .func
+        .signature
+        .params
+        .push(AbiParam::new(types::I32));
+    context
+        .func
+        .signature
+        .returns
+        .push(AbiParam::new(types::I32));
+
+    // Debug name for the function, not required, usually ExternalName::User is used instead.
+    context.func.name = cranelift::codegen::ir::ExternalName::TestCase {
+        length: 3,
+        ascii: [
+            b's', b'u', b'm', b'\0', b'\0', b'\0', b'\0', b'\0', b'\0', b'\0', b'\0', b'\0', b'\0',
+            b'\0', b'\0', b'\0',
+        ],
+    };
```

Comment:
This can use `ExternalName::testcase` and just pass a string instead of spelling out all the bytes.